### ISSUE TITLE
Object Pooling - Server null reference exception when spawning a pooled player object during SceneCondition

### ISSUE THREAD

**Issue by djgaven588** [FirstGearGames/FishNet#371]
**General**
Unity version: 2022.1.23f1
Fish-Networking version: 3.6.9
Link on our Discord where you troubleshot the issue(on Discord click the three dots next to the message, then Copy Message Link): https://discord.com/channels/424284635074134018/1034477094731784302/1107190412516544562

**Description**
When using object pooling with multiple player objects, accessing the Owner property in SceneCondition, retrieving NetworkObject.Owner produces a NullReferenceException. NetworkObject is not null at this point in execution however, and Owner should not return null or throw an exception. When destroying the player objects instead of pooling them, this issue does not occur.

**Replication**
1. Run a server build with object pooling and the scene condition. Using the following methods:
  - `NetworkObject nob = networkManager.GetPooledInstantiated(prefab, prefab.SpawnableCollectionId, true);`
  - `ServerManager.Spawn(character.NetworkObject, conn);`
2. Run 2 clients, and connect both to the server.
3. Disconnect 1 client, using the following method to pool the player:
- `ServerManager.Despawn(character.NetworkObject, DespawnType.Pool);`
4. Reconnect the client that you previously disconnected
5. Observe the exception produced on the server

**Expected behavior**
I expect that no exception should be thrown in this case, as this prevents the player object from initializing correctly, and the object will self destruct instead of pool once you try to despawn the object.

The stack trace can be found here: [Temp.txt](https://github.com/FirstGearGames/FishNet/files/11473037/Temp.txt). Line numbers may not perfectly match due to debugging attempts made to try and solve this issue ourselves. Our source code is available at https://github.com/jimdroberts/FishMMO, which is able to reproduce this consistently.

**Comment by FirstGearGames** [comment]
More likely the NetworkObject is null. Thanks for the info will check it out.

**Comment by FirstGearGames** [comment]
Looks like the observer condition is getting destroyed when it shouldn't be.

**Comment by FirstGearGames** [comment]
Please open NetworkObject.cs and goto this method 
`
internal void Deinitialize(bool asServer)
`
Replace the TRUE for FALSE here
`
NetworkObserver.Deinitialize(true);
`

**Comment by djgaven588** [comment]
@FirstGearGames We can't seem to find the line of code you are mentioning, and can't seem to find it in the repository here either. Would you be able to give more details on it?

**Comment by FirstGearGames** [comment]
I do see it's missing in 3.6.9. I've been working a bit on the NetworkObserver component so must of added that in afterwards. I cannot advise on what to do because in 3.7 I cannot replicate, perhaps I already fixed the issue during the rework. I'll send you an experimental package.

**Comment by djgaven588** [comment]
The 3.7 experimental does indeed have this fix.

The issue being that the observers are not being reinitialized, so their NetworkObjects are in an unreliable state.

I'll close this since it's already fixed, it's just not available yet.

For those that run into the issue while waiting for 3.7 release, change the following code to get back up and running in the mean time.
NetworkObserver.cs, line 202, add the following code after the if statement.
```cs
else
{
    for (int i = 0; i < _observerConditions.Count; i++)
    {
        _observerConditions[i].Initialize(networkObject);
    }
}
```

**Comment by djgaven588** [comment]
Oh actually there's a tag for this.. So I'll just leave it open so it can be tagged and closed on release.

**Comment by FirstGearGames** [comment]
Resolved in 3.7.0
